src/transaction_values.ts

```diff
@@ -83,113 +110,166 @@ async function resolveValue(t: Transaction, receipt: any, events: number) {
+                  const from = resolveLogAddress(log.topics[1]);
+                  const to = resolveLogAddress(log.topics[2]);
+                  const value = resolveLogValue(log.data);
+                  if (areEqualAddresses(from, t.to_address)) {
+                    totalValue += value;
+                    if (areEqualAddresses(to, royaltiesAddress)) {
+                      totalRoyalties += value;
+                    }
+                  }
+                }
+              } catch (e) {
+                logger.error(
+                  `Error adding royalties for transaction ${t.transaction}`,
+                  e
+                );
+              }
+            }
+          }
+        })
+      );
+      if (totalValue) {
+        t.value = totalValue;
+        t.royalties = totalRoyalties;
       }
-    });
+    }
   }
 
   return t;
 }
 
-export const runValues = async () => {
+const parseSeaportLog = async (
+  t: Transaction,
+  royaltiesAddress: string,
+  log: ethers.providers.Log
+) => {
+  let seaResult;
+  try {
+    seaResult = SEAPORT_IFACE.parseLog(log);
+  } catch (err: any) {
+    // logger.error('SEAPORT PARSE ERROR', t.transaction, err);
+    return null;
+  }
+
+  let recipientConsideration = seaResult.args.consideration?.find((c: any) =>
+    areEqualAddresses(c.recipient, t.from_address)
+  );
+  if (!recipientConsideration) {
+    recipientConsideration = seaResult.args.offer?.find((o: any) =>
+      areEqualAddresses(o.recipient, t.from_address)
+    );
+  }
+
+  const royaltiesConsideration = seaResult.args.consideration?.find((c: any) =>
+    areEqualAddresses(c.recipient, royaltiesAddress)
+  );
+
+  let tokenConsideration = seaResult.args.consideration?.find((o: any) =>
+    areEqualAddresses(o.token, t.contract)
+  );
+  if (!tokenConsideration) {
+    tokenConsideration = seaResult.args.offer?.find((o: any) =>
+      areEqualAddresses(o.token, t.contract)
+    );
+  }
+
+  if (tokenConsideration && recipientConsideration) {
+    const contract = tokenConsideration.token;
+    const tokenId = fromHex(tokenConsideration.identifier);
+    const royaltiesAmount = royaltiesConsideration
+      ? parseFloat(Utils.formatEther(royaltiesConsideration.amount))
+      : 0;
+
+    let totalAmount = 0;
+
+    seaResult.args.offer
+      .filter((o: any) => !areEqualAddresses(o.token, t.contract))
+      .map((o: any) => {
+        totalAmount += parseFloat(Utils.formatEther(o.amount));
+      });
+
+    if (totalAmount == 0) {
+      seaResult.args.consideration
+        .filter((o: any) => !areEqualAddresses(o.token, contract))
+        .map((o: any) => {
+          totalAmount += parseFloat(Utils.formatEther(o.amount));
+        });
+    }
+
+    return {
+      contract,
+      tokenId,
+      royaltiesAmount,
+      totalAmount
+    };
+  }
+};
+
+// HELPER FUNCTION FOR DEBUGGING VALUES USING TRX HASHES FROM DB
+export const debugValues = async () => {
```

Comment:
@TarmoKalling 
this is just a debug version which i use to resolve transaction values when needed - i renamed and addd a comment

about the 'transactions' array further down - it's just a collection of sample transactions 

tbh this should be a test....